src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureFilter.h

```diff
@@ -160,19 +160,18 @@ namespace OpenMS
       @param transitions transitions from a TargetedExperiment
 
       @return Map of labels/transition types and their corresponding number.
-    */ 
+    */
     std::map<String,int> countLabelsAndTransitionTypes(const Feature & component_group,
       const TargetedExperiment & transitions);
-    
+
     /**
-      @brief Sorts, removes duplicates, and concatenates a list of Strings
+      @brief Sorts the messages and returns a copy without duplicates
 
-      @param str_vec vector of Strings
-      @param delim token to separate Strings in the list
+      @param[in] messages A StringList containing the failure messages
 
-      @return A concatenated string.
-    */ 
-    String uniqueJoin(std::vector<String>& str_vec, String& delim);
+      @return A copy of the input, without duplicates
+    */
+    StringList getUniqueSorted(const StringList& messages) const;
```

Comment:
if you want to optimize this a bit, use pass by value
`StringList getUniqueSorted(StringList messages) const;` and work on `messages` directly. This allows the compiler to use copy-elision if r-values are passed in.
Even more fancy is to actually pass in r-values at the call site using std::move (if that is safe -- i did not check).

Since this is a pretty generic function, you might want to expose it as one and put it in a more prominent place (e.g. StringUtils). But I have no strong feelings about that.

Reply:
Thanks for the hint, I did not think about possible rvalues.
The fact that the function is currently called without rvalues would give us 0 advantages, correct?
I cannot pass the argument with `std::move()` because the original vector is later used for its size. I would like to rewrite a good portion of this class, in future. I don't like the way it looks.

If people would like to have the utility in `StringUtils`, then I will write the "optimized" version in it. Otherwise feel free to merge as it is.

Reply:
I actually had changed the code by using a `std::map` (because it sorts and keeps element unique), but then I decided to keep the changes at a minimum, because I'd like to keep the rewrite for the future.
The "cannot std::move" is more of a bad side-effect of the current state of the class, than a good reason.